### PR TITLE
One Window Stats

### PR DESCRIPTION
**Pull request by mdfeist**
Fixed stats for _4_bet_str and added a view to show all the stats in one window. Would like to add the ability to show both versions of the HUD at the same time.

### CODE REVIEW

File: poker-spirit/hud.py [hunk 1]
```diff
@@ -12,14 +12,33 @@
...
+            for stat, game_val in stats.items():
+                new_stats[player][stat] = game_val + career_stats[player][stat]
+        return new_stats
 
-class PlayerWindowsManager:
+    def populate(self, game, career_stats_by_player):
+        pass
+
+    def destroy():
+        PokerStarsGameIO
```
Comment (solinas) [inline poker-spirit/hud.py:34]: this line looks fishy

File: poker-spirit/hud.py [hunk 2]
```diff
@@ -12,14 +12,33 @@
 from dotenv import load_dotenv
 
 
+class PlayerStatsWindow:
+    def merge_current_and_career_stats(self, game, career_stats):
```
Comment (ajstarna) [inline poker-spirit/hud.py:16]: seems like this method is the main thing we need in both classes? But doesn't use self. Lets make it just a function?

File: poker-spirit/hud.py (hunk 1 above)
Comment (ajstarna) [inline poker-spirit/hud.py:34]: weird?

File: poker-spirit/hud.py [hunk 3]
```diff
@@ -12,14 +12,33 @@
...
+        return new_stats
 
-class PlayerWindowsManager:
+    def populate(self, game, career_stats_by_player):
+        pass
+
+    def destroy():
+        PokerStarsGameIO
+
+class PlayerWindowsManager(PlayerStatsWindow):
```
Comment (ajstarna) [inline poker-spirit/hud.py:36]: i feel like the manager shouldn't inherit. to me the manager is a contain of multiple windows.

File: poker-spirit/hud.py [hunk 4]
```diff
@@ -104,14 +108,55 @@ def populate(self, game, career_stats_by_player):
         for player_name in list(self.player_windows.keys()):
             if player_name not in game.current_players:
                 del self.player_windows[player_name]
-                del self.player_career_windows[player_name]                
+                del self.player_career_windows[player_name]    
+
+class AllPlayerWindowManager(PlayerStatsWindow):
```
Comment (ajstarna) [inline poker-spirit/hud.py:113]: i dont agree that this is a "manager" is it? Maybe just "AllPlayerStatsWindow"?

File: poker-spirit/analyze_hands.py [hunk 5]
```diff
@@ -357,7 +357,7 @@ def _3_bet_str(game_stats, player_name):
     def _4_bet_str(game_stats, player_name):
         stats = game_stats[player_name]
         if stats['4_bet_opp'] > 0:        
-            return f'4-Bet = {stats["3_bet"]}/{stats["4_bet_opp"]} = {round(100*stats["4_bet"]/stats["4_bet_opp"], 1)}'
+            return f'4-Bet = {stats["4_bet"]}/{stats["4_bet_opp"]} = {round(100*stats["4_bet"]/stats["4_bet_opp"], 1)}'
```
Comment (ajstarna) [inline poker-spirit/analyze_hands.py:360]: good catch

**Review by ajstarna — CHANGES_REQUESTED** [review]

File: poker-spirit/hud.py (hunk 4 above)
Comment (ajstarna) [reply poker-spirit/hud.py:113]: looking at the whole code again maybe im wrong i dunno

**Review by ajstarna — APPROVED** [review]

**Pull request merged**